extensions/amp-form/0.1/amp-form.js

```diff
@@ -1442,13 +1471,34 @@ export class AmpForm {
   }
 }
 
+/**
+ * Returns all element who's form attribute is the `form`
+ * that match the selectors.
+ * These elements must be contained by the form or <input>s
+ * that are outside the form.
+ * @param {!HTMLFormElement} form
+ * @param {string} query
+ * @return {!Array<HTMLElement>}
+ */
+export function formElementsQuerySelectorAll(form, query) {
+  return Array.from(form.elements).filter((element) => {
+    return (
+      matches(element, query) &&
+      (form.contains(element) || element.tagName.toUpperCase() === 'INPUT')
```

Comment:
I'm not sure it has to be `INPUT`. There's `textarea`, etc.  Why this condition?

```suggestion
```

Reply:
Are other `form` input elements (i.e. `textarea`), allowed to be outside of the form? I thought that wasn't the case, and that this PR is changing that too allow `inputs` to be outside now.

Reply:
Yep, all native input elements should be supported. This includes `input`, `textarea`, `select` and maybe others I'm forgetting.

I believe the `.form`/`.elements` relationship is all you need to confirm, so removing the check as per my earlier suggestion should be the way to go.